### PR TITLE
all: remove ineffectual assignments

### PR DESCRIPTION
**Pull request by kortschak**
I have left cephes and amos as is.

Please take a look.

### CODE REVIEW

File: lapack/testlapack/dgesvd.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func svdCheckPartial(t *testing.T, impl Dgesvder, job lapack.SVDJob, errStr stri
 	if shortWork {
```
Comment (kortschak) [inline lapack/testlapack/dgesvd.go]: Note for this file that there are ineffasigns to `jobU` and `jobV` that I don't understand how to resolve. @btracey do you have suggestions?

Comment (vladimir-ch) [reply lapack/testlapack/dgesvd.go]: Please wait with this, I've completely rewritten the Dgesvd test which I will submit soon.

Comment (kortschak) [reply lapack/testlapack/dgesvd.go]: I'll drop this commit.

**Comment by kortschak** [comment]
I'll wait for @vladimir-ch as well.

**Pull request merged**